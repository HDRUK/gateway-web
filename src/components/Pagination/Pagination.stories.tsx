import { Typography } from "@mui/material";
import type { Meta, StoryObj } from "@storybook/react";
import { useState } from "react";
import Pagination from "./Pagination";

const meta: Meta<typeof Pagination> = {
    component: Pagination,
};

export default meta;

type Story = StoryObj<typeof Pagination>;

const DummyComponent = () => {
    const [pageNumber, setPageNumber] = useState(1);

    const changeHandler = (_: unknown, page: number) => {
        setPageNumber(page);
    };
    return (
        <>
            <Typography sx={{ marginBottom: "10px" }}>
                Page number {pageNumber}
            </Typography>
            <Pagination
                count={5}
                onChange={changeHandler}
                variant="outlined"
                shape="rounded"
            />
        </>
    );
};

export const PaginationComponent: Story = {
    render: () => <DummyComponent />,
};
