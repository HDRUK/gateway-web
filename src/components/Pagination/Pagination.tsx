import { ArrowLeftIcon, ArrowRightIcon } from "@/consts/icons";
import MuiPagination, {
    PaginationProps as MuiPaginationProps,
} from "@mui/material/Pagination";
import PaginationItem from "@mui/material/PaginationItem";

interface PaginationProps extends MuiPaginationProps {
    isLoading?: boolean;
}

const Pagination = ({ isLoading, ...rest }: PaginationProps) => {
    if (isLoading) return null;
    return (
        <MuiPagination
            renderItem={item => (
                <PaginationItem
                    data-testid="pagination-item"
                    slots={{ previous: ArrowLeftIcon, next: ArrowRightIcon }}
                    {...item}
                />
            )}
            {...rest}
        />
    );
};

Pagination.defaultProps = {
    isLoading: false,
};

export default Pagination;
