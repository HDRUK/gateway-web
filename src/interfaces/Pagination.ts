interface PaginationType<T> {
    lastPage: number;
    to: number;
    from: number;
    currentPage: number;
    total: number;
    list: T[];
}

export type { PaginationType };
