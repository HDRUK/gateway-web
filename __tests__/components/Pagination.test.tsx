import { render, fireEvent } from "@testing-library/react";
import Pagination from "@/components/Pagination";

describe("Pagination", () => {
    it("renders without crashing", () => {
        const wrapper = render(<Pagination count={10} onChange={() => null} />);
        expect(wrapper.container).toMatchSnapshot();
    });

    it("calls onChange callback", () => {
        const handleChange = jest.fn();
        const { getByText } = render(
            <Pagination count={10} onChange={handleChange} />
        );

        const pageTwoButton = getByText("2");
        fireEvent.click(pageTwoButton);

        expect(handleChange).toHaveBeenCalledWith(expect.any(Object), 2);
    });
    it("should return empty div if still loading", () => {
        const wrapper = render(
            <Pagination count={10} isLoading onChange={() => null} />
        );

        expect(wrapper.container).toMatchSnapshot();
    });
});
