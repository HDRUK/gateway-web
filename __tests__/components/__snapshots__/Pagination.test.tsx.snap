// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders without crashing 1`] = `
<div>
  <nav
    aria-label="pagination navigation"
    class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
  >
    <ul
      class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
    >
      <li>
        <button
          aria-label="Go to previous page"
          class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled MuiPaginationItem-previousNext css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="ArrowLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m14 7-5 5 5 5V7z"
            />
          </svg>
        </button>
      </li>
      <li>
        <button
          aria-current="true"
          aria-label="page 1"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-selected MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          1
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 2"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          2
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 3"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          3
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 4"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          4
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to page 5"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          5
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <div
          class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
        >
          â€¦
        </div>
      </li>
      <li>
        <button
          aria-label="Go to page 10"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-page css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          10
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
      <li>
        <button
          aria-label="Go to next page"
          class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-rounded MuiPaginationItem-previousNext css-19xm0h7-MuiButtonBase-root-MuiPaginationItem-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
            data-testid="ArrowRightIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="m10 17 5-5-5-5v10z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </li>
    </ul>
  </nav>
</div>
`;
